wasm2c/wasm-rt.h

```diff
@@ -264,22 +264,6 @@ void wasm_rt_allocate_table(wasm_rt_table_t*,
  */
 void wasm_rt_free_table(wasm_rt_table_t*);
 
-#ifdef _WIN32
-float wasm_rt_truncf(float x);
-double wasm_rt_trunc(double x);
-float wasm_rt_nearbyintf(float x);
-double wasm_rt_nearbyint(double x);
-float wasm_rt_fabsf(float x);
-double wasm_rt_fabs(double x);
-#else
-#define wasm_rt_truncf(x) truncf(x)
-#define wasm_rt_trunc(x) trunc(x)
-#define wasm_rt_nearbyintf(x) nearbyintf(x)
-#define wasm_rt_nearbyint(x) nearbyint(x)
-#define wasm_rt_fabsf(x) fabsf(x)
-#define wasm_rt_fabs(x) fabs(x)
-#endif
```

Comment:
Is there some way we could keep these here in the shared header rather than making them part of the generated code?     Having them live in generator makes them a little hard to work on and make the diffs harder to read.

Also, I wonder if we could have a define to enable and disable them?  For example `#ifdef WASM2C_NON_CONFORMING_FLOAT_BITS` so something like that.  Then folks who don't care about the determinism of the nan bits can make a faster/slimmer build?

Reply:
Sure, they could live anywhere, but it defeats inlining if they are part of the runtime -- my working assumption was that we wanted to reserve runtime hooks for things where (a) different embedders might want to implement different behavior at runtime, or (b) there's some necessary whole-runtime state, e.g. `wasm_rt_register_func_type`. Whereas these functions seem about as universal as `popcount`, `ctz`, `trunc_sat`, `divrem`, etc., and I think will benefit from inlining or at least the compiler being able to see the implementation.

How would you feel about doing this as a command-line argument to wasm2c, e.g., `--non-conforming-float-bits` ? I could imagine a similar flag for `--non-conforming-memory-loads` (which would remove the load liveness stuff at the cost of sometimes omitting a load that would have produced a deterministic trap).

(BTW if you want the template to be easier to work on, we _could_ split it into separate files for the `%%includes` and `%%declarations` sections, and then both of those would become fully-valid C files. And we could omit the `src/prebuilt/wasm2c.include.*` files (or even fac.c) from the repository and just always build them as part of the build system, which would make the diffs cleaner in Git.)

Reply:
If we are going to have flags for controlling spec compliance do you think they should be wasm2c arguments that control codegen, or should we emit code with `#ifdef` and make these compile time options?   I'm leaning towards that latter but I don't feel strongly and you have done more work on wasm2c recently, so I think we should pick the one that makes maintenance of wasm2c easier.

Reply:
I don't have a strong view -- honestly the easiest is probably just to not do it until somebody asks for it, but second easiest is probably the `#ifdef` option. Just pushed this (with `WASM2C_NON_CONFORMING_NAN_BITS` as the symbol).